lib/rules/force-types-on-object-props.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * @author Przemysław Jan Beigert
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Helpers
+// ------------------------------------------------------------------------------
+
+/**
+ * Check if all keys and values from second object are resent in first object
+ *
+ * @param {{ [key: string]: any }} a object to
+ * @param {{ [key: string]: any }} b The string to escape.
+ * @returns {boolean} Returns the escaped string.
+ */
+const isLooksLike = (a, b) =>
+  a &&
+  b &&
+  Object.keys(b).every((bKey) => {
+    const bVal = b[bKey]
+    const aVal = a[bKey]
+    if (typeof bVal === 'function') {
+      return bVal(aVal)
+    }
+    return bVal == null || /^[bns]/.test(typeof bVal)
+      ? bVal === aVal
+      : isLooksLike(aVal, bVal)
+  })
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'enforce user to add type declaration to object props',
+      categories: ['base'],
+      recommended: false,
+      url: 'https://eslint.vuejs.org/rules/force-types-on-object-props.html'
+    },
+    fixable: null,
+    schema: []
+  },
+  /** @param {RuleContext} context */
+  create(context) {
+    return {
+      /** @param {ExportDefaultDeclaration} node */
+      ExportDefaultDeclaration(node) {
```

Comment:
Please have a look at https://github.com/vuejs/eslint-plugin-vue/blob/master/lib/rules/require-prop-types.js for how to find `props`. Your solution e.g. doesn't cover `<script setup>` or `Vue.extend({…})`.

Reply:
oh, it looks good. I'll try to rewrite my rule using these utils

Reply:
@FloEdelmann I did it. Much better